### ISSUE TITLE
HTML Comments with extra '-' parse incorrectly

### ISSUE THREAD

**Issue by djlauber** [doxygen/doxygen#7333]
Discovered after upgrading from 1.8.11 to 1.8.16 that using long HTML comments  as visual breaks in my markdown files caused all documentation after them to disappear.

## Minimum example
Create markdown file containing the following and build it into HTML documentation using Doxygen 1.8.16.
```
\mainpage Comment Demo
<!---->1
<!-- -->2
<!--A-->3
<!-- --->4
-->5
<!--- -->6
<!---- -->7
<!----- -->8
<!--a a-->9
```
### Expected Result
`1 2 3 4 -->5 6 7 8 9`
### Actual Result
`<!-—>1 2 3 5 <!— -->6 <!-— -->7 8 9`

## Issues
**Seems to indicate 2 problems:**
- Doesn't recognize closing with more than 2 consecutive dashes _(introduced between 1.8.11 and 1.8.16)_
- Doesn't recognize opening with 3 or 4 consecutive dashes _(not sure when introduced)_

**Comment by albert-github** [comment]
The good news,the problem already solved in the current master of doxygen due to pull request #7245, so in the master this results in: `1 2 3 5 6 7 8 9 `
Problem that looks like to remain is is the missing `4 -->` between the 3 and the 5, but actually this is not a problem as "An end comment cannot contain 3 - signs" (in  XML in a nustshell , chapter 2 about comments,it is stated: "In particular, a three-hyphen close like `-->` is specifically forbidden" and the "end"  of 4 has 3 hyphens.

So the problem is fixed but not released.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.17.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
